librFSM/CMakeLists.txt

```diff
@@ -79,6 +79,15 @@ install(TARGETS rFSM
         ARCHIVE DESTINATION lib
         PUBLIC_HEADER DESTINATION include)
 
+include(CMakePackageConfigHelpers)
+install(EXPORT rFSM
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/rFSM)
```

Comment:
It is good practice to have a `NAMESPACE` argument to try to avoid target names collisions, see https://pabloariasal.github.io/2018/02/19/its-time-to-do-cmake-right/#this-is-how-you-do-cmake .

Reply:
Ok. `NAMESPACE` added (db2ae50). But unfortunately, I had to comment out `add_subdirectory(examples)` into the main `CMakeLists.txt` because otherwise, I retrieved the following error.
```
CMake Warning (dev) at examples/CMakeLists.txt:16 (add_executable):
  Policy CMP0028 is not set: Double colon in target name means ALIAS or
  IMPORTED target.  Run "cmake --help-policy CMP0028" for policy details.
  Use the cmake_policy command to set the policy and suppress this warning.

  Target "rfsmTest" links to target "rFSM::rFSM" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
This warning is for project developers.  Use -Wno-dev to suppress it.
```
```
/usr/bin/ld: cannot find -lrFSM::rFSM
collect2: error: ld returned 1 exit status
```
 I think is due to the fact that when the example is compiled the `rFSM` library is not installed yet.

Reply:
The typical strategy to deal with this is to add an appropriate `ALIAS` also for making sure that `rFSM::rFSM` is available also during the build phase, see https://github.com/robotology/how-to-export-cpp-library/blob/e23e399fce102ba8f87cd5269797513000b99794/src/LibTemplateCMake/CMakeLists.txt#L62 . 
I would recommenced anyhow to use a `NAMESPACE` consistent with the package name (in your case `rFSMTools`).